engine/src/main/java/org/terasology/rendering/dag/stateChanges/SetFboWriteMask.java

```diff
@@ -83,21 +83,21 @@ public StateChange getDefaultInstance() {
 
     @Override
     public int hashCode() {
-        return Objects.hashCode(fbo.fboId);
+        return Objects.hashCode(fbo);
     }
 
     @Override
     public boolean equals(Object obj) {
         return (obj instanceof SetFboWriteMask)
-                && fbo.fboId == ((SetFboWriteMask) obj).fbo.fboId
+                && fbo == ((SetFboWriteMask) obj).fbo
```

Comment:
Hmmm. Confusing issue. 

Ok, for the way you changed it, two instances of SetFboWriteMask must hold a reference to the exact same FBO to be considered the same - never mind the mask for a moment.

_Theorethically_ however, there could be two FBO objects that are distinct but point to the same fbo GPU-side - they store the same fbo id. I don't think Terasology's code allows this so it's a very remote theory. But in that circumstance the two state changes would be considered different even though the FBOs could, in fact, be the same.

In short I agree that we should change the == fbo object comparison to fbo.equals(obj), if that's what you meant.